src/linesearch/line_model.jl

```diff
@@ -53,7 +60,8 @@ i.e.,
 """
 function grad(f::LineModel, t::AbstractFloat)
   NLPModels.increment!(f, :neval_grad)
-  return dot(grad(f.nlp, f.x + t * f.d), f.d)
+  @. f.xt = f.x + t * f.d
+  return dot(grad(f.nlp, f.xt), f.d)
```

Comment:
This method should call `grad!()`.

Reply:
I would need to store `g` inside `LineModel` as well, is that what you mean?